src/parseTree.js

```diff
@@ -281,37 +270,121 @@
     props: {
       traceNode: {type: Object, required: true},
       measureInputText: {type: Function},
-      classes: {type: Object, default: Object},
       isInVBox: {type: Boolean},
-      vbox: {type: Boolean}
+
+      // from parent element
+      context: {type: Object},
+
+      // from currentLR
+      currentLR: {type: Object, default: Object},
+      eventHandlers: {type: Object}
     },
     computed: {
+      labeled: function() {
+        return shouldNodeBeLabeled(this.traceNode);
+      },
+      vbox: function() {
+        return hasVisibleChoice(this.traceNode) ||
+               hasVisibleLeftRecursion(this.traceNode) ||
+               (isAlt(this.traceNode.expr) && this.context && this.context.vbox);
+      },
+      isWhitespace: function() {
+        return this.traceNode.ruleName === 'spaces';
+      },
+      isLeaf: function() {
+        var leaf = isLeaf(this.traceNode);
+        if (this.traceNode.isMemoized) {
+          var memoKey = this.traceNode.expr.toMemoKey();
+          var stack = this.currentLR[memoKey];
+          if (stack && stack[stack.length - 1] === this.traceNode.pos) {
+            leaf = true;
+          }
+        }
+        return leaf;
+      },
       classObj: function() {
         var obj = {
-          disclosure: this.classes.labeled && this.isInVBox
+          disclosure: this.labeled && this.isInVBox
         };
         var ctorName = this.traceNode.ctorName;
         if (ctorName) {
           obj[ctorName.toLowerCase()] = true;
         }
-        Object.assign(obj, this.classes);
+
+        obj.collapsed = this.labeled &&
```

Comment:
I think we should do this differently. The `collapsed` property is _exactly_ what Vue.js `data` is for -- for view state that can change and affects how the view is rendered. To make the children appear, it should be as easy as `this.collapsed = true` down in `setCollapsed`.

Here's what I'd try:

- Define a computed property named `initiallyCollapsed`, using the same logic you're using here.
- Then, add `data: function() { return {collapsed: false}; }` to make a reactive `collapsed` property.
- In `created()` and `updated()`, do `this.collapsed = this.initiallyCollapsed` to set the default value based on the value of `initiallyCollapsed` (unfortunately I don't think there's a better way to do this)

Then, setting `this.collapsed = false` should trigger a re-render.

I don't think you need `childrenNodes` in data -- that's why `children` is a computed property, so you can return either the children nodes or `[]` depending on the value of `this.collapsed`.